lib/import.js

```diff
@@ -0,0 +1,157 @@
+
+  console.log( 'Mapping data'.header )
+  // Map all the old shabad IDs to new ones
+  const shabadIDMap = shabadIds.reduce( ( ids, id, index ) => ( {
+    ...ids,
+    [ id ]: newShabadIDs[ index ],
+  } ), {} )
+
+  // Map sources to original shabad id
+  const sourceMap = lines.reduce( ( sourceMap, line ) => ( {
+    ...sourceMap,
+    [ line[ shabadId ] ]: sources.find( source => source === line[ source ] ) || -1,
+  } ), {} )
+
+  // Generate sources
+  const newSources = sources.map( ( name, index ) => ( {
+    name_english: `${name}-new`,
+    name_gurmukhi: `${name}-new`,
+    id: newSourceId + index,
+    length: -1,
+    page_name_english: -( index + 1 ),
+    page_name_gurmukhi: -( index + 1 ),
+  } ) )
+
+  // Generate new Shabads
+  const newShabads = shabadIds.map( ( id, index ) => ( {
+    id: shabadIDMap[ id ],
+    writer_id: -1,
+    section_id: -1,
+    sttm_id: lines[ sttmId ],
+    order_id: shabadOrderId + index + 1,
+    source_id: sourceMap[ id ],
+  } ) )
+
+  // Generate the new lines
+  const newLines = lines.map( ( data, index ) => ( {
+    id: newLineIDs[ index ],
+    shabad_id: shabadIDMap[ data[ shabadId ] ],
+    gurmukhi: data[ gurmukhi ],
+    source_page: data[ page ] || -1,
+    source_line: data[ line ] || -1,
+    order_id: lineOrderId + 1 + index,
+  } ) )
+
+  // Generate translation sources for translations
+  const newTranslationSources = translation.map( ( name, index ) => ( {
+    id: newTranslationSourceId + index,
+    name_gurmukhi: name,
+    name_english: name,
+    source_id: newSourceId,
+    language_id: -1,
+  } ) )
+
+  // Generate the translations
+  const newTranslations = translation
+    .map( ( name, index ) => lines.map( ( data, lineIndex ) => ( {
+      line_id: newLineIDs[ lineIndex ],
+      translation_source_id: newTranslationSourceId + index,
+      translation: data[ name ] || '',
+      additional_information: '{}',
+    } ) ) )
+    .reduce( ( allTranslations, translations ) => allTranslations.concat( translations ), [] )
+
+  // Now, insert all the data
+  console.log( 'Inserting into SQLite database'.header )
+  await knex.transaction( async trx => {
+    // Insert sources
+    await Promise.all( newSources.map( source => Sources.query( trx ).insert( source ) ) )
+    // Insert shabad IDs
+    await Promise.all( newShabads.map( shabad => Shabads.query( trx ).insert( shabad ) ) )
+    // Insert lines
+    await Promise.all( newLines.map( line => Lines.query( trx ).insert( line ) ) )
+    // Insert translation sources
+    await Promise.all( newTranslationSources.map( source => (
+      TranslationSources.query( trx ).insert( source )
+    ) ) )
+    // Insert translations
+    await Promise.all( newTranslations.map( data => Translations.query( trx ).insert( data ) ) )
+  } )
+
+  console.log( 'Import complete. Please update the default values in `build/database.sqlite`. Run `npm run build-json`, followed by `npm run build-sqlite`.'.success )
```

Comment:
Would it make sense to add checks for `-1` when running these `npm run build` commands?

Reply:
What do you mean by `-1`?

Reply:
oh, sorry, I understand. Could be worth it, yeah.
Will add an issue.